pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go

```diff
@@ -65,6 +67,11 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
+	// if Workload.Type is not empty, means componentdefinition refer to an already existing workloaddefinition
+	if componentDefinition.Spec.Workload.Type != "" {
```

Comment:
we also need to generate openAPI and docs in this case

Reply:
Got it